### PR TITLE
Rename Plex variable fallbackurl to localurl

### PR DESCRIPTION
**Pull request by twolaw**
In `.env` file, I suggest renaming the variable `PLEX_FALLBACKURL` to `PLEX_LOCALURL` so that **users understand and change this URL by themself** with a specific message in error logs.

______
After that, the local IP address auto-finding will have to be improved for docker users because 172.17.0.x doesn't look useful (see #775 and #816)

### CODE REVIEW

File: plextraktsync/plex_server.py [hunk 1]
```diff
@@ -58,7 +60,9 @@ def _get_plex_server():
                 # save new url to .env
                 CONFIG["PLEX_TOKEN"] = plex_token
                 CONFIG["PLEX_BASEURL"] = new_plex_baseurl
-                CONFIG["PLEX_FALLBACKURL"] = plex_fallbackurl
+                plex_localurl = CONFIG.get("PLEX_LOCALURL")
```
Comment (glensc) [inline plextraktsync/plex_server.py:63]: have you tested does `CONFIG.get("PLEX_LOCALURL")` thow if key is missing? maybe need `CONFIG.get("PLEX_LOCALURL", None)`

**Comment by glensc** [comment]
In my mind there should be a change in config.py as well, adding PLEX_LOCALURL environment extraction.

I would probably make fallback to previous variable in config class, not in the outside (plex-server.py) doing similar checks multiple times.

**Comment by glensc** [comment]
also, atomic commits. renaming things and adding fallback should have been two separate commits.

**Comment by glensc** [comment]
also added "docs needed" label, as this should be probably documented somewhere at some point.

File: plextraktsync/plex_server.py (hunk 1 above)
Comment (twolaw) [reply plextraktsync/plex_server.py:63]: yes i tested this

**Comment by twolaw** [comment]
Thanks for your comments, I improved the code.
I think it is useless to mention this in readme file because average user will never have to go deaply into this configuration problem.
The error message is enough imho and now explicitly says : `Plex connection error: {}, local url {} didn't respond either. Check PLEX_LOCALURL in .env file.`

Maybe the error message should also talk about `plex_login` command ?

**Review by glensc — APPROVED** [review]

**Comment by glensc** [comment]
Ok. I like the new version more (approved). But I can't test this myself.

**Comment by twolaw** [comment]
tested successfully, I merge

**Comment by keldian** [comment]
I'm noticing that when both BASEURL and LOCALURL are satisfied, PTS still listens on the BASEURL value. Would it not be more efficient to listen on the local host by default?

**Comment by twolaw** [comment]
> Would it not be more efficient to listen on the local host by default?

Using localhost by default could not work with Plex instances requiring SSL and with shared libraries.

The PLEX_LOCALURL value is not populated by Plex but by the script. Therefore it is not 100% reliable.
The PLEX_BASEURL is provided by the Plex server itself and uses SSL which is much more reliable.

You can edit both PLEX_BASEURL and PLEX_LOCALURL as you wish in your .env file (exchange values if you prefer).

**Comment by glensc** [comment]
What about saving all plex urls that plex-login is able to detect and sort them somehow? I think plex python library itself sorts them already by availability or speed

**Comment by twolaw** [comment]
It can be a good idea to help users with dynamic ip address.

> sorts them already by availability or speed

and also local status and protocol (http vs https).

```xml
https://plex.tv/pms/resources.xml?includeHttps=1&X-Plex-Token=xxxxxxxxxxxxxxxxxxxxx

<Connection protocol="https" address="192.168.0.10" port="32400" uri="https://192-168-0-10.b25b53d5e99c40bc94126387dxxxxxxx.plex.direct:32400" local="1"/>
<Connection protocol="https" address="176.16.x.x" port="16422" uri="https://176-16-x-x.b25b53d5e99c40bc94126387dxxxxxxx.plex.direct:16422" local="0"/>
```

**Pull request merged**